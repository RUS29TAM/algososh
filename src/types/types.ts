import {ElementStates} from "./element-states";

export interface ICircleDetail {
    symbol?: null | string;
    state: ElementStates;
}

export interface IColumn {
    number: number;
    state: ElementStates;
}
