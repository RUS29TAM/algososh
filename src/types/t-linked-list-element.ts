export type TLinkedListElement = {
    isAddHead: boolean;
    isDeleteHead: boolean;
    isAddTail: boolean;
    isDeleteTail: boolean;
    isAddByIndex: boolean;
    isDeleteByIndex: boolean;
}
