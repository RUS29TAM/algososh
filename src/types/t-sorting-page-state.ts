export type TSortingPageState = {
    isIncreasing: boolean;
    isDecreasing: boolean;
    isLoading: boolean;
}
