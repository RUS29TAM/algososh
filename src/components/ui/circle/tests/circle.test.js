import renderer from 'react-test-renderer'
import {Circle} from "../circle";
import {ElementStates} from "../../../../types/element-states";

describe('Тест для компонента Circle', () =>{
    it('Компонент Circle отрисовка компонента с символами', () => {
        const circle = renderer.create(<Circle letter={'symbol'}/>).toJSON();
        expect(circle).toMatchSnapshot();
    })

    it('Компонент Circle отрисовка без символов', () => {
        const circle = renderer.create(<Circle letter={''}/>).toJSON();
        expect(circle).toMatchSnapshot();
    })

    it('Компонент Circle отрисовка head', () => {
        const circle = renderer.create(<Circle head={'string'}/>).toJSON();
        expect(circle).toMatchSnapshot();
    })

    it('Компонент Circle отрисовка tail', () => {
        const circle = renderer.create(<Circle tail={'string'}/>).toJSON();
        expect(circle).toMatchSnapshot();
    })

    it('Компонент Circle отрисовка head c react-element', () => {
        const circle = renderer.create(<Circle head={<Circle isSmall={true}/>} />).toJSON();
        expect(circle).toMatchSnapshot();
    })

    it('Компонент Circle отрисовка tail c react-element', () => {
        const circle = renderer.create(<Circle tail={<Circle isSmall={true}/>} />).toJSON();
        expect(circle).toMatchSnapshot();
    })

    it('Компонент Circle отрисовка по индексу', () => {
        const circle = renderer.create(<Circle index={0}/>).toJSON();
        expect(circle).toMatchSnapshot();
    })

    it('Компонент Circle отрисовка с реквизитом isSmall', () => {
        const circle = renderer.create(<Circle isSmall={true}/>).toJSON();
        expect(circle).toMatchSnapshot();
    })

    it('Компонент Circle отрисовка в состоянии Default', () => {
        const circle = renderer.create(<Circle state={ElementStates.Default}/>).toJSON();
        expect(circle).toMatchSnapshot();
    })

    it('Компонент Circle отрисовка в состоянии Changing', () => {
        const circle = renderer.create(<Circle state={ElementStates.Changing}/>).toJSON();
        expect(circle).toMatchSnapshot();
    })

    it('Компонент Circle отрисовка в состоянии Modified', () => {
        const circle = renderer.create(<Circle state={ElementStates.Modified}/>).toJSON();
        expect(circle).toMatchSnapshot();
    })
})
