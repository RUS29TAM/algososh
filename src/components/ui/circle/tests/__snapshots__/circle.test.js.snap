// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тест для компонента Circle Компонент Circle отрисовка head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    string
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тест для компонента Circle Компонент Circle отрисовка head c react-element 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle  small default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тест для компонента Circle Компонент Circle отрисовка tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    string
  </div>
</div>
`;

exports[`Тест для компонента Circle Компонент Circle отрисовка tail c react-element 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle  small default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
</div>
`;

exports[`Тест для компонента Circle Компонент Circle отрисовка без символов 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тест для компонента Circle Компонент Circle отрисовка в состоянии Changing 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тест для компонента Circle Компонент Circle отрисовка в состоянии Default 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тест для компонента Circle Компонент Circle отрисовка в состоянии Modified 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тест для компонента Circle Компонент Circle отрисовка компонента с символами 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      symbol
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тест для компонента Circle Компонент Circle отрисовка по индексу 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    0
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`Тест для компонента Circle Компонент Circle отрисовка с реквизитом isSmall 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;
